### ISSUE TITLE
Add `params` and `query` to `useRouter` and `withRouter`

### ISSUE THREAD

**Issue by flybayer** [blitz-js/blitz#623]
### What do you want and why?

#574 added new `useParams` and `useRouterQuery` hooks that separate route params from url query items.

So now we need to update the router object from `useRouter` and `withRouter` to have the same behavior.

**Current:**
```tsx
// Everything lumped in query
const {query} = useRouter()
```

**Change to:**
```tsx
// Separate route params from query string items
const {params, query} = useRouter()
```

For extra context, see #530 

This would be a breaking change for existing blitz apps

### Possible implementation(s)

Currently we reexport `useRouter` and `withRouter` from Next.js. We need to add our own `useRouter` and `withRouter` functions that re-use the ones from Next, but set the `params` and `query` keys as shown above. 

You can use the new `useParams` and `useRouterQuery` hooks to get the data for those keys.

**Comment by toshi1127** [comment]
I want to work in this, Can I take this issue?

**Comment by flybayer** [comment]
@toshi1127 you sure can! :)

**Comment by flybayer** [comment]
Closed via #677
